### PR TITLE
x/tokenfactory: .proto files

### PR DESCRIPTION
**Pull request by sunnya97**
## What is the purpose of the change

*Add a description of the overall background and high level changes that this PR introduces *

This PR adds the proto files relevant for the tokenfactory module.

It contains the following Msg types for the module:
- CreateDenom
- MintTo
- BurnFrom
- ChangeAdmin

It contains the following queries:
- DenomAuthorityMetadata
- DenomsFromCreator

It contains a commented out Msg type for "force transfers".  Force transfers are disabled based on discussion in: https://github.com/osmosis-labs/osmosis/issues/1029

## Testing and Verifying

This change is only proto file additions without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: proto/osmosis/tokenfactory/v1beta1/authorityMetadata.proto [hunk 1]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+package osmosis.tokenfactory.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v7/x/tokenfactory/types";
+
+message DenomAuthorityMetadata {
```
Comment (ValarDragon) [inline proto/osmosis/tokenfactory/v1beta1/authorityMetadata.proto:9]:
```suggestion
// DenomAuthorityMetadata specifies metadata for addresses that have specific capabilities over a token factory denom.
// Right now there is only one Admin permission, but is planned to be extended to the future.
message DenomAuthorityMetadata {
```

File: proto/osmosis/tokenfactory/v1beta1/genesis.proto [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+message GenesisDenom {
+  option (gogoproto.equal) = true;
+
+  string denom = 1 [ (gogoproto.moretags) = "yaml:\"denom\"" ];
+  DenomAuthorityMetadata authority_metadata = 2 [
+    (gogoproto.moretags) = "yaml:\"authority_metadata\"",
+    (gogoproto.nullable) = false
+  ];
+}
```
Comment (ValarDragon) [inline proto/osmosis/tokenfactory/v1beta1/genesis.proto:27]: Why is this type called genesis denom? Presumably this is whats going to get stored in state for every created denom (as the admin metadata has to be maintained somewhere)

Comment (ValarDragon) [reply proto/osmosis/tokenfactory/v1beta1/genesis.proto:27]: Perhaps it should be called FactoryDenom?

Comment (sunnya97) [reply proto/osmosis/tokenfactory/v1beta1/genesis.proto:27]: It's only the type that's stored in the genesis file.  In state, `denom` is a key and just `DenomAuthorityMetadata` is serialized and stored in state

**Review by ValarDragon — APPROVED** [review]

File: proto/osmosis/tokenfactory/v1beta1/tx.proto [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+//     (gogoproto.moretags) = "yaml:\"amount\"",
+//     (gogoproto.nullable) = false
+//   ];
+//   string transferFromAddress = 3
+//       [ (gogoproto.moretags) = "yaml:\"transfer_from_address\"" ];
+//   string transferToAddress = 4
+//       [ (gogoproto.moretags) = "yaml:\"transfer_to_address\"" ];
+// }
+
+// message MsgForceTransferResponse {}
```
Comment (alexanderbez) [inline proto/osmosis/tokenfactory/v1beta1/tx.proto:70]: Do we need this commented out?

File: proto/osmosis/tokenfactory/v1beta1/tx.proto [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  rpc CreateDenom(MsgCreateDenom) returns (MsgCreateDenomResponse);
+  rpc Mint(MsgMint) returns (MsgMintResponse);
+  rpc Burn(MsgBurn) returns (MsgBurnResponse);
+  // ForceTransfer is deactivated for now because we need to think through edge
+  // cases rpc ForceTransfer(MsgForceTransfer) returns
+  // (MsgForceTransferResponse);
+  rpc ChangeAdmin(MsgChangeAdmin) returns (MsgChangeAdminResponse);
+}
+
+// ===================== MsgCreateDenom
```
Comment (alexanderbez) [inline proto/osmosis/tokenfactory/v1beta1/tx.proto:20]: nit: IMO I find these comment blocks unhelpful and kinda annoying. Can we just stick with simple docs above the type?

```suggestion
```

**Review by alexanderbez — APPROVED** [review]
Thanks for breaking this up @sunnya97!

File: proto/osmosis/tokenfactory/v1beta1/tx.proto (hunk 3 above)
Comment (sunnya97) [reply proto/osmosis/tokenfactory/v1beta1/tx.proto:70]: Wanted to deactivate this message for now, but we will eventually add it back in in the future

File: proto/osmosis/tokenfactory/v1beta1/genesis.proto (hunk 2 above)
Comment (p0mvn) [inline proto/osmosis/tokenfactory/v1beta1/genesis.proto:27]: nit: newline

File: proto/osmosis/tokenfactory/v1beta1/query.proto [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    (gogoproto.nullable) = false
+  ];
+}
+
+message QueryDenomsFromCreatorRequest {
+  string creator = 1 [ (gogoproto.moretags) = "yaml:\"denoms\"" ];
+}
+message QueryDenomsFromCreatorResponse {
+  repeated string denoms = 1 [ (gogoproto.moretags) = "yaml:\"denoms\"" ];
+}
```
Comment (p0mvn) [inline proto/osmosis/tokenfactory/v1beta1/query.proto:41]: nit: missing newline

File: proto/osmosis/tokenfactory/v1beta1/tx.proto (hunk 3 above)
Comment (p0mvn) [reply proto/osmosis/tokenfactory/v1beta1/tx.proto:78]: Should we track this in an issue?

**Review by p0mvn — APPROVED** [review]

**Pull request merged**